src/server/main_service.cc

```diff
@@ -2201,6 +2215,13 @@ void Service::Pubsub(CmdArgList args, ConnectionContext* cntx) {
     PubsubChannels(pattern, cntx);
   } else if (subcmd == "NUMPAT") {
     PubsubPatterns(cntx);
+  } else if (subcmd == "NUMSUB") {
+    if (args.size() == 1) {
+      (*cntx)->SendEmptyArray();
+      return;
+    }
```

Comment:
This isn't really needed, `PubsubNumSub()` will gracefully handle an empty list

Reply:
fixed: https://github.com/dragonflydb/dragonfly/pull/2282/commits/d553f071fe61ded3376352dbc7d08e2da292f5dd